storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -130,14 +130,15 @@ public BasicContainer(ContainerType type, Map<String, Object> conf, String super
         super(type, conf, supervisorId, supervisorPort, port, assignment,
             resourceIsolationManager, workerId, topoConf, ops, metricsRegistry, containerMemoryTracker);
         assert (localState != null);
+        assert (resourceIsolationManager != null);
```

Comment:
instead of assert (which will not be active at runtime), do we want to raise an IllegalArgumentException here? If resourceIsolationManager is null, then there will be NPE anyway elsewhere.

Reply:
This will not really happen since `resourceIsolationManager` has default value and will be always initialized. `assert` is just used at development phase to avoid errors. We can continue our discussion on `assert` at the other pr (https://github.com/apache/storm/pull/3274), but I can remove this line if you prefer.

Reply:
We can continue assert discussion elsewhere. But my preference is not to add this.